model/PermissionsService.php

```diff
@@ -111,6 +134,17 @@ private function getActions(array $resourcesToUpdate, array $permissionsList, ar
             }
         }
 
+        // It is likely the strategy has not checked for duplicate permissions,
+        // so we need to remove them here.
+        //
+        foreach ($actions['add'] as &$entry) {
+            $entry['permissions'] = array_unique($entry['permissions']);
```

Comment:
Keep in mind that array_unique does not work with matrix https://www.php.net/manual/en/function.array-unique.php#refsect1-function.array-unique-notes

Reply:
thanks